### PR TITLE
Implement collapse action

### PR DESCRIPTION
**Pull request by sdwolfz**
As mentioned here: https://github.com/Alexander-Miller/treemacs/issues/534#issuecomment-810565109

This provides a `COLLAPSE` action that defined the behavior of a key that navigates up the directory structure. Together with the keybind changes allow for one handed navigation using `hjkl` (similar to nerdtree/neotree).

### CODE REVIEW

**Comment by sdwolfz** [comment]
@Alexander-Miller should be ready for review now.

File: README.org [hunk 1]
```diff
@@ -13,37 +13,37 @@
 
 * Content                                                                            :TOC:noexport:
 - [[#quick-feature-overview][Quick Feature Overview]]
-   - [[#fancy-gifs][Fancy Gifs!]]
```
Comment (Alexander-Miller) [inline README.org:16]: Any idea why the indent is changed? I thought setting the `org-list-indent-offset` in the header line should take care of that.

File: src/extra/treemacs-evil.el [hunk 2]
```diff
@@ -97,8 +97,8 @@ Then call ORIG-FUN with its ARGS and reopen treemacs if it was open before."
 (evil-define-key 'treemacs treemacs-mode-map (kbd "gr")     #'treemacs-refresh)
 (evil-define-key 'treemacs treemacs-mode-map [down-mouse-1] #'treemacs-leftclick-action)
 (evil-define-key 'treemacs treemacs-mode-map [drag-mouse-1] #'treemacs-dragleftclick-action)
-(evil-define-key 'treemacs treemacs-mode-map (kbd "h")      #'treemacs-root-up)
-(evil-define-key 'treemacs treemacs-mode-map (kbd "l")      #'treemacs-root-down)
+(evil-define-key 'treemacs treemacs-mode-map (kbd "h")      #'treemacs-COLLAPSE-action)
+(evil-define-key 'treemacs treemacs-mode-map (kbd "l")      #'treemacs-RET-action)
```
Comment (Alexander-Miller) [inline src/extra/treemacs-evil.el:102]: LGTM. Add a short entry in the Changelog.org and we're good to go.

**Review by Alexander-Miller — CHANGES_REQUESTED** [review]

File: README.org (hunk 1 above)
Comment (sdwolfz) [reply README.org:16]: No idea, I tried with both loading, and not loading settings, `toc-org-max-depth: 2` gets applied, but the offset does not... I'll edit it in fundamental mode and push.

File: src/extra/treemacs-evil.el (hunk 2 above)
Comment (sdwolfz) [reply src/extra/treemacs-evil.el:102]: Sure, btw, why does `treemacs-evil` need this particular overwrite here, but only for the `h` and `l` keys? I couldn't figure it out...

**Comment by Alexander-Miller** [comment]
:+1:

File: src/extra/treemacs-evil.el (hunk 2 above)
Comment (Alexander-Miller) [reply src/extra/treemacs-evil.el:102]: treemacs-evil-state derives from motion-state, so anything already bound there needs to be overridden here.

File: README.org (hunk 1 above)
Comment (Alexander-Miller) [reply README.org]: org-toc is always using the global value. I've opened a PR for it: https://github.com/snosov1/toc-org/pull/75

**Pull request merged**